sdk/metric/aggregator/gauge/gauge_test.go

```diff
@@ -94,3 +94,58 @@ func TestGaugeMonotonicDescending(t *testing.T) {
 		require.Equal(t, first, agg.AsNumber(), "Same last value - monotonic")
 	})
 }
+
+func TestGaugeNormalMerge(t *testing.T) {
+	ctx := context.Background()
+
+	test.RunProfiles(t, func(t *testing.T, profile test.Profile) {
+		agg1 := New()
+		agg2 := New()
+
+		batcher, record := test.NewAggregatorTest(export.GaugeMetricKind, profile.NumberKind, false)
+
+		first1 := profile.Random(+1)
+		first2 := profile.Random(+1)
+		first1.AddNumber(profile.NumberKind, first2)
+
+		agg1.Update(ctx, first1, record)
+		agg2.Update(ctx, first2, record)
+
+		agg1.Collect(ctx, record, batcher)
+		agg2.Collect(ctx, record, batcher)
+
+		t1 := agg1.Timestamp()
+		t2 := agg2.Timestamp()
+		require.True(t, t1.Before(t2))
+
+		agg1.Merge(agg2, record.Descriptor())
+
+		require.Equal(t, t2, agg1.Timestamp(), "Merged values - non-monotonic")
+		require.Equal(t, first2, agg1.AsNumber(), "Merged values - non-monotonic")
+	})
+}
+
+func TestGaugeMonotonicMerge(t *testing.T) {
+	ctx := context.Background()
+
+	test.RunProfiles(t, func(t *testing.T, profile test.Profile) {
+		agg1 := New()
+		agg2 := New()
+
+		batcher, record := test.NewAggregatorTest(export.GaugeMetricKind, profile.NumberKind, true)
+
+		first1 := profile.Random(+1)
+		agg1.Update(ctx, first1, record)
+
+		first2 := profile.Random(+1)
+		first2.AddNumber(profile.NumberKind, first1)
+		agg2.Update(ctx, first2, record)
+
+		agg1.Collect(ctx, record, batcher)
+		agg2.Collect(ctx, record, batcher)
+
+		agg1.Merge(agg2, record.Descriptor())
+
+		require.Equal(t, first2, agg1.AsNumber(), "Merged values - monotonic")
```

Comment:
Do we ignore checking timestamp in the monotonic case on purpose?

Reply:
Added another test here.